From 53c328099693339d9826ad8802bb156b2be6f9fb Mon Sep 17 00:00:00 2001
From: Philipp Reinkemeier <sigsegv@philipp-r.de>
Date: Wed, 21 Jan 2026 20:41:40 +0100
Subject: [PATCH] Fixed an issue in lddmc_makenode and lddmc_make_copynode.   -
 Both methods take two MDDs as argument, and call lddmc_refs_push on     both
 of them, but lddmc_refs_pop is called with 1 reference to pop     instead of
 2.

---
 3rd_party/sylvan/src/sylvan_ldd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/3rd_party/sylvan/src/sylvan_ldd.c b/3rd_party/sylvan/src/sylvan_ldd.c
index b8883fd..8a58a44 100644
--- a/3rd_party/sylvan/src/sylvan_ldd.c
+++ b/3rd_party/sylvan/src/sylvan_ldd.c
@@ -367,7 +367,7 @@ lddmc_makenode(uint32_t value, MDD ifeq, MDD ifneq)
         lddmc_refs_push(ifeq);
         lddmc_refs_push(ifneq);
         RUN(sylvan_gc);
-        lddmc_refs_pop(1);
+        lddmc_refs_pop(2);
 
         index = llmsset_lookup(nodes, n.a, n.b, &created);
         if (index == 0) {
@@ -394,7 +394,7 @@ lddmc_make_copynode(MDD ifeq, MDD ifneq)
         lddmc_refs_push(ifeq);
         lddmc_refs_push(ifneq);
         RUN(sylvan_gc);
-        lddmc_refs_pop(1);
+        lddmc_refs_pop(2);
 
         index = llmsset_lookup(nodes, n.a, n.b, &created);
         if (index == 0) {
-- 
2.52.0

